packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -677,76 +743,54 @@ export const createNodesForContentType = ({
             }
 
-            entryItemFields[`${entryItemFieldKey}___NODE`] = jsonNodeId
-            delete entryItemFields[entryItemFieldKey]
-          } else if (
-            fieldType === `Object` &&
-            _.isArray(entryItemFields[entryItemFieldKey])
-          ) {
-            entryItemFields[`${entryItemFieldKey}___NODE`] = []
-
-            entryItemFields[entryItemFieldKey].forEach((obj, i) => {
-              const jsonNodeId = createNodeId(
-                `${entryNodeId}${entryItemFieldKey}${i}JSONNode`
-              )
-
-              // The Contentful model has `.sys.updatedAt` leading for an entry. If the updatedAt value
-              // of an entry did not change, then we can trust that none of its children were changed either.
-              // (That's why child nodes use the updatedAt of the parent node as their digest, too)
-              const existingNode = getNode(jsonNodeId)
-              if (existingNode?.updatedAt !== entryItem.sys.updatedAt) {
-                const jsonNode = prepareJSONNode(
-                  jsonNodeId,
-                  entryNode,
-                  entryItemFieldKey,
-                  obj
-                )
-                childrenNodes.push(jsonNode)
-              }
-
-              entryItemFields[`${entryItemFieldKey}___NODE`].push(jsonNodeId)
-            })
-
-            delete entryItemFields[entryItemFieldKey]
-          }
-        })
+            entryItemFields[`${entryItemFieldKey}___NODE`].push(jsonNodeId)
+          })
 
-        entryNode = {
-          ...entryItemFields,
-          ...entryNode,
-          node_locale: locale.code,
+          delete entryItemFields[entryItemFieldKey]
         }
+      })
 
-        // The content of an entry is guaranteed to be updated if and only if the .sys.updatedAt field changed
-        entryNode.internal.contentDigest = entryItem.sys.updatedAt
+      entryNode = {
+        ...entryItemFields,
+        ...entryNode,
+        node_locale: locale.code,
+      }
 
-        // Link tags
-        if (pluginConfig.get(`enableTags`)) {
-          entryNode.metadata = {
-            tags___NODE: entryItem.metadata.tags.map(tag =>
-              createNodeId(`ContentfulTag__${space.sys.id}__${tag.sys.id}`)
-            ),
-          }
+      // The content of an entry is guaranteed to be updated if and only if the .sys.updatedAt field changed
+      entryNode.internal.contentDigest = entryItem.sys.updatedAt
+
+      // Link tags
+      if (pluginConfig.get(`enableTags`)) {
+        entryNode.metadata = {
+          tags___NODE: entryItem.metadata.tags.map(tag =>
+            createNodeId(`ContentfulTag__${space.sys.id}__${tag.sys.id}`)
+          ),
         }
+      }
 
-        return entryNode
-      })
-      .filter(Boolean)
+      return entryNode
+    })
 
-    entryNodes.forEach(entryNode => {
-      createNodePromises.push(createNode(entryNode))
+    entryNodes.forEach((entryNode, index) => {
+      if (entryNode) {
```

Comment:
In what case can an entry node not exist?

Reply:
Oops, this was leftover from when I was debugging. I hardcoded a node limit at one point and I think that caused an error here. Removing this now